plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/BOAndDPCompileHook.java

```diff
@@ -52,6 +46,22 @@ public void validate(Map<String, Object> toCheck) throws Exception {
 
             File pluginDir = (File) moreInfo.get("pluginDir");
 
+            Path pluginSourcesDir = config.getLocalCheckoutDir().toPath();
+
+            if (pluginSourcesDir != null) {
+                boolean isMultipleLocalPlugins = config.getIncludePlugins() != null && config.getIncludePlugins().size() > 1;
+                // We are running for local changes, let's copy the .eslintrc file if we can
+                //If we are using localCheckoutDir with multiple plufins the .eslintrc must be located at the top level
+                // If not it must be located on the parent of the localCheckoutDir
```

Comment:
What should be the behaviour in case of setting the localCheckoutDir with only a set of BO plugins locally checked out? All of BO plugins that are not in the local directory will be checked out from SCM. Would be there any impact? Which .eslintrc file should be used?

Reply:
@fcojfernandez If localCheckoutDir is defined then the existing PCT hook will not try to checkout any BO plugin, resulting in the PCT to follow usual ways and fail when trying to checkout the missing plugins.

And I agree with that, mixing released versions of BO plugins with local ones does not make sense due to dependencies between them. Also note that this behaviour is not introduced by this change but is already present in the current PCT.

So, if using multiple local plugins the .eslintrc file used will be the one at the top of the localCheckoutDir (if present) if checking from SCM the .eslintrc file will be there as we are checking out the aggregator, and a mixed scenario IMO does not make sense and is going to fail before the build